src/Extensions/DependencyInjection/FunctionsHostBuilderContext.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.Functions.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Context containing the common services and settings for the Functions host.
+    /// </summary>
+    public abstract class FunctionsHostBuilderContext
+    {
+        private readonly WebJobsBuilderContext _webJobsBuilderContext;
+
+        public FunctionsHostBuilderContext(WebJobsBuilderContext webJobsBuilderContext)
+        {
+            _webJobsBuilderContext = webJobsBuilderContext ?? throw new ArgumentNullException(nameof(webJobsBuilderContext));
+        }
+
+        /// <summary>
+        /// Gets the name of the environment. This is "Development" when running in the Azure Functions Core Tools and "Production" when deployed to Azure.
+        /// </summary>
+        public string EnvironmentName => _webJobsBuilderContext.EnvironmentName;
```

Comment:
Would it be possible to expose `IHostEnvironment` here (which in turn exposes the EnvironmentName)? This gives parity with the generic host, whilst also being a little more flexible.

Reply:
That introduced a little more complexity. The `IHostEnvironment` interface changed between .NET Core 2 and 3 so we'd need to multi-target this project and deal with those differences. We decided to abstract that away and not directly expose the interface. The other properties were also confusing in the context of Functions as they refer to the location and name of the host on the VM, not your deployed application.

This value is exactly the `IHostEnvironment.EnvironmentName` -- it gets copied here by the host. If there are other values from `IHostEnvironment` that are useful, we can expose them here.